implementation/src/main/java/io/smallrye/config/WithUnnamedKey.java

```diff
@@ -0,0 +1,23 @@
+package io.smallrye.config;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Omits a single map key from the configuration name when populating {@link java.util.Map} types. Configuration values
+ * for the {@link java.util.Map} may be retrieved by the key defined in {@link WithUnnamedKey#value()}.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE_USE })
+public @interface WithUnnamedKey {
+    /**
+     * The key name to use to populate configuration values from a configuration path without a key.
+     *
+     * @return the Map key
+     */
+    String value() default "unnamed";
```

Comment:
Is there any way this could default to `null` instead? I mean an actual `null`, not the string `"null"` of course.

Asking because in Hibernate Search for example, we have a map of backend config, and the "default"/"unnamed" backend typically has its name set to `null` throughout all SPIs. So forcing a non-null key here leads to all sorts of conversions that introduce some fragility (if I forget to convert at some point, it will likely not work and might not be spotted by tests).

Reply:
I guess we could. This was to provide a more explicit lookup and avoid having to look for `null` key, but it should be fine.

Reply:
I've changed the behaviour to use the null key. Can you try it out?

Reply:
I updated my PR (https://github.com/quarkusio/quarkus/pull/32854) and I confirm null keys work fine. Thank you!

Reply:
Great!